src/cpp/shared/combat/army.cpp

```diff
@@ -1343,83 +1345,100 @@ void SpecialAttackGraphics(army *attacker, army *target) {
   gpCombatManager->DrawFrame(0, 1, 0, 1, 75, 1, 1);
 
-  int targMidX = target->MidX();
-  int targMidY = target->MidY();
-  if (attacker->creatureIdx == CREATURE_LICH || attacker->creatureIdx == CREATURE_POWER_LICH) {
-    targMidX = gpCombatManager->combatGrid[target->occupiedHex].centerX;
-    targMidY = gpCombatManager->combatGrid[target->occupiedHex].occupyingCreatureBottomY - 17;
-  }
-  int projStartX;
-  if (attacker->facingRight == 1)
-    projStartX = attacker->frameInfo.projectileStartOffset[1][0] + gpCombatManager->combatGrid[attacker->occupiedHex].centerX;
-  else
-    projStartX = gpCombatManager->combatGrid[attacker->occupiedHex].centerX - attacker->frameInfo.projectileStartOffset[1][0];
-  int projStartY = attacker->frameInfo.projectileStartOffset[1][1] + gpCombatManager->combatGrid[attacker->occupiedHex].occupyingCreatureBottomY;
-  int totXDiff = targMidX - projStartX;
-  char firingLeft = 0;
-  if (targMidX - projStartX < 0) {
-    firingLeft = 1;
-    totXDiff = -totXDiff;
+  int targMidX, targMidY;
+  int totXDiff, yDiff;
+  char firingLeft;
+  {
+    targMidX = target->MidX();
+    targMidY = target->MidY();
+    if (attacker->creatureIdx == CREATURE_LICH || attacker->creatureIdx == CREATURE_POWER_LICH) {
+      targMidX = gpCombatManager->combatGrid[target->occupiedHex].centerX;
+      targMidY = gpCombatManager->combatGrid[target->occupiedHex].occupyingCreatureBottomY - 17;
+    }
+    int projStartX;
+    if (attacker->facingRight == 1)
+      projStartX = attacker->frameInfo.projectileStartOffset[1][0] + gpCombatManager->combatGrid[attacker->occupiedHex].centerX;
+    else
+      projStartX = gpCombatManager->combatGrid[attacker->occupiedHex].centerX - attacker->frameInfo.projectileStartOffset[1][0];
+    int projStartY = attacker->frameInfo.projectileStartOffset[1][1] + gpCombatManager->combatGrid[attacker->occupiedHex].occupyingCreatureBottomY;
+    totXDiff = targMidX - projStartX;
+    firingLeft = 0;
+    if (targMidX - projStartX < 0) {
+      firingLeft = 1;
+      totXDiff = -totXDiff;
+    }
+    yDiff = targMidY - projStartY;
   }
-  int yDiff = targMidY - projStartY;
 
   float angleDeg;
   int spriteIdx;
-  if (totXDiff) {
-    float slope = (double)-yDiff / (double)totXDiff;
-    angleDeg = atan(slope) * 180.0 / 3.14159;
-    int i;
-    for (i = 1;  attacker->frameInfo.numMissileDirs > i &&
-      (*(float *)((char *)&attacker->frameInfo.projectileStartOffset[i + 2] + 1) +
-        attacker->frameInfo.projDirAngle[i]) / 2.0 >= angleDeg;  ++i)
-      ;
-    if (attacker->frameInfo.numMissileDirs <= i)
-      spriteIdx = attacker->frameInfo.numMissileDirs - 1;
-    else
-      spriteIdx = i - 1;
-  } else {
-    if (yDiff <= 0)
-      spriteIdx = 0;
-    else
-      spriteIdx = attacker->frameInfo.numMissileDirs - 1;
-    angleDeg = (double)(yDiff <= 0 ? 90 : -90);
+  {
+    if (totXDiff) {
+      float slope = (double)-yDiff / (double)totXDiff;
+      angleDeg = atan(slope) * 180.0 / 3.14159;
+      int i;
```

Comment:
This can be moved inside the loop I think

Reply:
Is there a need though?